In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

In [2]:
dataset=pd.read_csv("Dataset_for_Internship.csv",encoding='unicode_escape')
dataset.head(5)

Restaurant ID                Restaurant Name  Country Code       City  \
0       17284302  El Vaquero Mexican Restaurant           216     Albany   
1       17330155                       Deorio's           216   Columbus   
2       17334679                         Azteca           216  Davenport   
3       17342548  Happy Joe's Pizza & Ice Cream           216    Dubuque   
4       17582499                    Royal Hotel           216  Pocatello   

   Longitude  Latitude                    Cuisines  Average Cost for two  \
0   -84.2194   31.6158                     Mexican                     0   
1   -84.9427   32.4810              Italian, Pizza                     0   
2   -90.5683   41.5699                     Mexican                     0   
3   -90.7264   42.4955  Desserts, Pizza, Ice Cream                     0   
4  -112.0127   42.6192             Pizza, Bar Food                     0   

    Currency Has Table booking Has Online delivery Is delivering now  \
0  Dollar($)                No                  No                No   
1  Dollar($)                No                  No                No   
2  Dollar($)                No                  No                No   
3  Dollar($)                No                  No                No   
4  Dollar($)                No                  No                No   

   Price range  Aggregate rating Rating color Rating text  Votes  
0            1               3.4       Orange     Average     45  
1            1               4.0        Green   Very Good    170  
2            1               4.3        Green   Very Good    167  
3            1               3.5       Yellow        Good     74  
4            1               3.6       Yellow        Good     59

# Task 4 :Online Delivery

### 1.Determine the percentage of restaurants that offer online delivery.

In [3]:
total_restro_offer_online_delivery=dataset["Has Online delivery"].value_counts()
total_restro_offer_online_delivery

Has Online delivery
No     7091
Yes    2451
Name: count, dtype: int64

In [4]:
per_offer_online_delivery=((total_restro_offer_online_delivery/len(dataset))*100)
per_offer_online_delivery

Has Online delivery
No     74.313561
Yes    25.686439
Name: count, dtype: float64

The analysis shows that 74.31% of restaurants do not have online delivery, while 25.69% do offer it. This means most restaurants still depend on dine-in or takeaway, with only a smaller number offering the convenience of online delivery, suggesting there is room for more restaurants to start offering delivery services.

 ### 2.Compare the average ratings of restaurants with and without online delivery.

In [5]:
average_rating_with_delivery = dataset[dataset['Has Online delivery'] == "Yes"]['Aggregate rating'].mean()
average_rating_without_delivery = dataset[dataset['Has Online delivery'] == "No"]['Aggregate rating'].mean()
print(f"{average_rating_with_delivery} Has online delivery and {average_rating_without_delivery} Dont have online delivery")

3.2488372093023252 Has online delivery and 2.463517134395713 Dont have online delivery


Restaurants that offer online delivery have an average rating of 3.25, while those that don't offer online delivery have a lower average rating of 2.46. This suggests that restaurants with online delivery tend to receive better customer ratings, possibly due to the added convenience and accessibility of their services.